fabriziogianni7/botblock
# feat(protect): fetch robots.txt from internet

## Description

The first step for the protect flow will be to get the content's creator already created `robots.txt`. On a follow up PR, we will be able to support copy-pasting if they don't have their site up but for now this will suffice.

## Implementation

Since most of the `robots.txt` are protected by some CORS policy, we added a simple express server that does the request for us and is available for the app to access.

On the app's side of the feature, we just created a `useRobots` hook that does the proper request and returns the body of the `robots.txt` file.

## Next steps

We shall create a React Context for robots, so we can persist the fetched value across components and use it accordingly.

Additionally, we will create a proper landing screen for the protect flow as defined in:
- #2 

# How to test

1. Start express server by running `yarn start` on `./packages/express`
2. Add the `useRobots(url)` call on some screen and define the value for the url as you see fit
3. Start nextjs by running `yarn dev` on `./packages/nextjs`
4. Browse to that page and validate that the feature is working

## Related links

- Closes #1
